analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -413,5 +413,171 @@ public void Invocation_IsNullOrEmpty_TryFinally()
                 new SymbolicValue().WithConstraint(ObjectConstraint.NotNull)
             });
         }
+
+        [DataTestMethod]
+        [DynamicData(nameof(ThrowHelperCalls))]
+        public void Invocation_ThrowHelper_StopProcessing(string throwHelperCall)
+        {
+            var code = $@"
+Tag(""Before"");
+{throwHelperCall}
+Tag(""Unreachable"");
+";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder("Before");
+            validator.ValidateExitReachCount(0);
+            validator.ValidateExecutionCompleted();
+        }
+
+        [DataTestMethod]
+        [DynamicData(nameof(ThrowHelperCalls))]
+        public void Invocation_ThrowHelper_OnlyInBranch(string throwHelperCall)
+        {
+            var code = @$"
+if (condition)
+{{
+    Tag(""Before"");
+    {throwHelperCall}
+    Tag(""Unreachable"");
+}}
+Tag(""End"");
+";
+            var validator = SETestContext.CreateCS(code, ", bool condition").Validator;
+            validator.ValidateTagOrder("Before", "End");
+            validator.ValidateExitReachCount(1);
+            validator.ValidateExecutionCompleted();
+        }
+
+        [DataTestMethod]
+        [DynamicData(nameof(ThrowHelperCalls))]
+        public void Invocation_ThrowHelper_TryCatchFinally(string throwHelperCall)
+        {
+            var code = @$"
+try
+{{
+    {throwHelperCall}
+    Tag(""Unreachable"");
+}}
+catch
+{{
+    Tag(""Catch"");
+}}
+finally
+{{
+    Tag(""Finally"");
+}}
+Tag(""End"");
+";
+            var validator = SETestContext.CreateCS(code, ", bool condition").Validator;
+            validator.ValidateTagOrder("Catch", "Finally", "Finally", "End");
+            validator.ValidateExitReachCount(2);
+            validator.ValidateExecutionCompleted();
+        }
+
+        [DataTestMethod]
+        [DataRow("System.Diagnostics.CodeAnalysis.DoesNotReturn")]
+        [DataRow("OtherNamespace.WithAttributeSuffix.DoesNotReturn")]
```

Comment:
Based on the namespace name, shouldn't the name be like this?
```suggestion
        [DataRow("OtherNamespace.WithAttributeSuffix.DoesNotReturnAttribute")]
```

Reply:
In this case, both names are valid.

Reply:
Yes, both are valid. But the UTs seems wrong to me. Originally, we had this pair
```
        [DataRow("OtherNamespace.WithAttributeSuffix.DoesNotReturn")]
        [DataRow("OtherNamespace.WithoutAttributeSuffix.DoesNotReturn")]
```
And I'm looking at the `WithAttributeSuffix` vs. `WithoutAttributeSuffix` declared intention (I like these namespaces). 

The point is, that `Without` is tested here
```
        [DataRow("OtherNamespace.WithoutAttributeSuffix.DoesNotReturn")]
                                                                     ^ No suffix here - all good
```
but the `With` didn't have it
```
        [DataRow("OtherNamespace.WithAttributeSuffix.DoesNotReturn")]
                                                                  ^ No suffix here either. And it should be IMO
```
So currently, we have these 3
```
        [DataRow("OtherNamespace.WithAttributeSuffix.DoesNotReturn")]   // <--- this one doesn't make sense, it's tested on the 3rd line
        [DataRow("OtherNamespace.WithAttributeSuffix.DoesNotReturnAttribute")]    // OK
        [DataRow("OtherNamespace.WithoutAttributeSuffix.DoesNotReturn")]          // OK
```

Reply:
...aaand now I see it, it's with suffix or without suffix on the class declaration.

While doable, I don't see a need to support `class DoesNotReturn`. Attributes should have `Attribute` suffix. And those we really need do have it. Let's remove it to simplify the logic.